components/new_post/post_component.tsx

```diff
@@ -0,0 +1,332 @@
+import Tooltip from 'components/tooltip';
+import PostProfilePicture from 'components/post_profile_picture';
+import FailedPostOptions from 'components/post_view/failed_post_options';
+import PostAriaLabelDiv from 'components/post_view/post_aria_label_div';
+import PostTime from 'components/post_view/post_time';
+import ReactionList from 'components/post_view/reaction_list';
+import MessageWithAdditionalContent from 'components/message_with_additional_content';
+import InfoSmallIcon from 'components/widgets/icons/info_small_icon';
+import PostPreHeader from 'components/post_view/post_pre_header';
+import EditPost from 'components/edit_post';
+import AutoHeightSwitcher, {AutoHeightSlots} from 'components/common/auto_height_switcher';
+import {Props as TimestampProps} from 'components/timestamp/timestamp';
+
+import PostUserProfile from './user_profile';
+import PostOptions from './post_options';
+
+export type Props = {
+    post: Post;
+    teamId: string;
+    currentUserId: string;
+    compactDisplay?: boolean;
+    colorizeUsernames?: boolean;
+    isFlagged: boolean;
+    isBusy?: boolean;
+    removePost: (post: Post) => void;
+    previewCollapsed?: string;
+    previewEnabled?: boolean;
+    isEmbedVisible?: boolean;
+    enableEmojiPicker: boolean;
+    enablePostUsernameOverride: boolean;
+    isReadOnly: boolean;
+    pluginPostTypes?: {[postType: string]: PostPluginComponent};
+    channelIsArchived?: boolean;
+    isConsecutivePost?: boolean;
+    isLastPost?: boolean;
+    recentEmojis: Emoji[];
+    handleCardClick?: (post: Post) => void;
+    togglePostMenu?: (opened: boolean) => void;
+    a11yIndex?: number;
+    isBot: boolean;
+    actions: {
+        markPostAsUnread: (post: Post, location: string) => void;
+        emitShortcutReactToLastPostFrom: (emittedFrom: 'CENTER' | 'RHS_ROOT' | 'NO_WHERE') => void;
+        setActionsMenuInitialisationState: (viewed: Record<string, boolean>) => void;
+        selectPost: (post: Post) => void;
+    };
+    timestampProps?: Partial<TimestampProps>;
+    shouldHighlight?: boolean;
+    isPostBeingEdited?: boolean;
+    isMobileView: boolean;
+};
+
+const PostComponent = (props: Props): JSX.Element => {
+    const postRef = useRef<HTMLDivElement>(null);
+    const postHeaderRef = useRef<HTMLDivElement>(null);
+    const [hover, setHover] = useState(false);
+    const [a11yActive, setA11y] = useState(false);
+    const [fileDropdownOpened, setFileDropdownOpened] = useState(false);
+    const [alt, setAlt] = useState(false);
+
+    useEffect(() => {
+        if (postRef.current) {
+            postRef.current.addEventListener(A11yCustomEventTypes.ACTIVATE, handleA11yActivateEvent);
+            postRef.current.addEventListener(A11yCustomEventTypes.DEACTIVATE, handleA11yDeactivateEvent);
+        }
+        if (a11yActive) {
+            postRef.current?.dispatchEvent(new Event(A11yCustomEventTypes.UPDATE));
+        }
+    }, [postRef.current]);
```

Comment:
i dont think ref can be passed as dependency, since mutating them wont trigger render